developer/js/index.ts

```diff
@@ -232,6 +162,91 @@ export default class LexicalModelCompiler {
   };
 
+  /**
+   * Returns the generated code for the model that will ultimately be loaded by
+   * the LMLayer worker. This code contains all model parameters, and specifies
+   * word breakers and auxilary functions that may be required.
+   * 
+   * @param model_id      The model ID. TODO: not sure if this is actually required!
+   * @param modelSource   A specification of the model to compile
+   * @param sourcePath    Where to find auxilary sources files
+   */
+  generateLexicalModelCode(model_id: string, modelSource: LexicalModelSource, sourcePath: string) {
+    let oc: LexicalModelCompiled = {id: model_id, format: modelSource.format};
+
+    let sources: string[] = modelSource.sources.map(function(source) {
+      return fs.readFileSync(path.join(sourcePath, source), 'utf8');
+    });
+
+    // TODO: add metadata in comment
+    const filePrefix: string = `(function() {\n'use strict';\n`;
+    const fileSuffix: string = `})();`;
+    let func = filePrefix;
+
+    let wordBreakingSource: string = null;
+
+    // Figure out what word breaker the model is using, if any.
+    if (modelSource.wordBreaking) {
+      if (typeof modelSource.wordBreaking === "string") {
+        // It must be a builtin word breaker, so just instantiate it.
+        wordBreakingSource = `wordBreakers['${modelSource.wordBreaking}']`;
+      } else if (modelSource.wordBreaking.sources) {
+        let wordBreakingSources: string[] = modelSource.wordBreaking.sources.map(function(source) {
+          return fs.readFileSync(path.join(sourcePath, source), 'utf8');
+        });
+
+        wordBreakingSource = this.transpileSources(wordBreakingSources).join('\n');
+      }
+    }
+
+    //
+    // Emit the model as code and data
+    //
+
+    switch(modelSource.format) {
+      case "custom-1.0":
+        func += this.transpileSources(sources).join('\n');
+        func += `LMLayerWorker.loadModel(new ${modelSource.rootClass}());\n`;
+        break;
+      case "fst-foma-1.0":
+        (oc as LexicalModelCompiledFst).fst = Buffer.from(sources.join('')).toString('base64');
+        this.logError('Unimplemented model format '+modelSource.format);
+        return false;
+      case "trie-1.0":
+        func += `var model = {};\n`;
+        func += `model.backingData = ${createWordListDataStructure(sources)};\n`;
+        func += `LMLayerWorker.loadModel(new models.WordListModel(model.backingData`;
+        if (wordBreakingSource) {
+          func += `, {wordBreaking: ${wordBreakingSource}}`;
+        }
+        func += `));\n`;
+        break;
+      case 'trie-2.0':
+        func += `LMLayerWorker.loadModel(new models.TrieModel(${
+          createTrieDataStructure(sources, modelSource.searchTermToKey)
+        }, {\n`;
+        if (wordBreakingSource) {
+          func += `  wordBreaking: ${wordBreakingSource},\n`;
+        }
+        if (modelSource.searchTermToKey) {
+          func += `  searchTermToKey: ${modelSource.searchTermToKey.toString()},\n`;
```

Comment:
I see we're assuming that `searchTermToKey` will be an inline-defined function within the model's base declaration?  I suppose that's fine as long as it's simple enough - I tested it out with a simple dummy function to ensure compilation worked.

Reply:
Yup! If the need arises to make this more complicated later, we can always augment to this definition.